src/app/pages/MarginTradePage/components/PositionEventsTable/PositionEventRow.tsx

```diff
@@ -2,45 +2,77 @@ import React, { useMemo } from 'react';
 import { useTranslation } from 'react-i18next';
 import { translations } from 'locales/i18n';
 import { LinkToExplorer } from 'app/components/LinkToExplorer';
-import { weiToAssetNumberFormat } from 'utils/display-text/format';
-import { AssetRenderer } from 'app/components/AssetRenderer';
-import { PositionEvent } from '.';
 import { DisplayDate } from 'app/components/ActiveUserLoanContainer/components/DisplayDate';
+import { assetByTokenAddress } from 'utils/blockchain/contract-helpers';
+import { bignumber } from 'mathjs';
+import { AssetValue } from 'app/components/AssetValue';
+import { toWei } from 'utils/blockchain/math-helpers';
+import {
+  CloseWithSwap,
+  DepositCollateral,
+  Liquidate,
+  Trade,
+} from 'utils/graphql/rsk/generated';
+import { EventType } from '../../types';
 
 type LiquidatedPositionRowProps = {
-  event: PositionEvent;
-  positionStatus: boolean;
+  event: CloseWithSwap | DepositCollateral | Liquidate | Trade;
   isLong: boolean;
+  positionToken: string;
+  collateralToken: string;
+  positionSize: string;
+  closePrice?: string;
 };
 
 export const PositionEventRow: React.FC<LiquidatedPositionRowProps> = ({
   event,
+  positionToken,
+  collateralToken,
+  positionSize,
+  closePrice,
+  isLong,
 }) => {
   const { t } = useTranslation();
-  const timestamp = useMemo(() => new Date(event.time).getTime().toString(), [
-    event,
-  ]);
+  const timestamp = useMemo(
+    () => new Date(event.timestamp).getTime().toString(),
+    [event],
+  );
+  const loanAsset = assetByTokenAddress(positionToken);
+  const collateralAsset = assetByTokenAddress(collateralToken);
+
+  const exitPrice = useMemo(() => {
+    if (!closePrice) {
+      return false;
+    }
+    if (event.__typename === EventType.LIQUIDATE) {
+      return closePrice;
+    }
+    return isLong ? bignumber(1).div(closePrice).toString() : closePrice;
+  }, [closePrice, isLong, event]);
 
   return (
     <tr>
-      <td>{t(translations.tradeEvents[event.event])}</td>
+      <td>
+        {event.__typename ? t(translations.tradeEvents[event.__typename]) : '-'}
+      </td>
       <td className="tw-hidden md:tw-table-cell">
         <div className="tw-whitespace-nowrap">
-          {event.positionSubtracted && <>-</>}
-          {weiToAssetNumberFormat(
-            event.positionChange,
-            event.positionAsset,
-          )}{' '}
-          <AssetRenderer asset={event.positionAsset} />
+          {event.__typename === EventType.LIQUIDATE ? '-' : ''}
```

Comment:
Extract `event.__typename === EventType.LIQUIDATE` to a condition (e.g. `isLiquidateEvent`) and use `useMemo`.